samples/dnn/virtual_try_on.py

```diff
@@ -0,0 +1,386 @@
+    input_image = (input_image - 127.5) / 127.5
+    input_image = input_image.transpose(2, 0, 1)
+
+    img_head = input_image * phead - (1 - phead)
+
+    from PIL import Image
+    pose_shape = pose_shape.astype(np.uint8).reshape(height, width)
+    parse_shape = Image.fromarray(pose_shape)
+    parse_shape = parse_shape.resize((width // 16, height // 16), Image.BILINEAR)
+    parse_shape = parse_shape.resize((width, height), Image.BILINEAR)
+    res_shape = np.array(parse_shape)
+
+    res_shape = cv.dnn.blobFromImage(res_shape, 1.0 / 127.5, mean=(127.5, 127.5, 127.5), swapRB=True)
+    res_shape = res_shape.squeeze(0)
+
+    agnostic = np.concatenate((res_shape, img_head, pose_map), axis=0)
+    agnostic = np.expand_dims(agnostic, axis=0)
+    return agnostic
+
+
+def get_warped_cloth(cloth_path, agnostic, model, backend, target):
+    cloth_img = cv.imread(cloth_path)
+    cloth = cv.dnn.blobFromImage(cloth_img, 1.0 / 127.5, mean=(127.5, 127.5, 127.5), swapRB=True)
+
+    theta = run_gmm(agnostic, cloth, model, backend, target)
+    grid = postprocess(theta)
+    warped_cloth = bilinear_sampler(cloth, grid).astype(np.float32)
+    return warped_cloth
+
+
+def run_gmm(agnostic, c, model, backend, target):
+    class CorrelationLayer(object):
+        def __init__(self, params, blobs):
+            super(CorrelationLayer, self).__init__()
+
+        def getMemoryShapes(self, inputs):
+            fetureAShape = inputs[0]
+            b, c, h, w = fetureAShape
+            return [[b, h * w, h, w]]
+
+        def forward(self, inputs):
+            feature_A, feature_B = inputs
+            b, c, h, w = feature_A.shape
+            feature_A = feature_A.transpose(0, 1, 3, 2)
+            feature_A = np.reshape(feature_A, (b, c, h * w))
+            feature_B = np.reshape(feature_B, (b, c, h * w))
+            feature_B = feature_B.transpose(0, 2, 1)
+            feature_mul = feature_B @ feature_A
+            feature_mul= np.reshape(feature_mul, (b, h, w, h * w))
+            feature_mul = feature_mul.transpose(0, 1, 3, 2)
+            correlation_tensor = feature_mul.transpose(0, 2, 1, 3)
+            correlation_tensor = np.ascontiguousarray(correlation_tensor)
+            return [correlation_tensor]
+
+    cv.dnn_registerLayer('Correlation', CorrelationLayer)
+    net = cv.dnn.readNet(model)
+
+    net.setInput(agnostic, "input.1")
+    net.setInput(c, "input.18")
+    net.setPreferableBackend(backend)
+    net.setPreferableTarget(target)
+    theta = net.forward()
+
+    cv.dnn_unregisterLayer('Correlation')
+    return theta
+
+
+def get_tryon(agnostic, warp_cloth, model, backend, target):
+    net = cv.dnn.readNet(model)
+    inp = np.concatenate([agnostic, warp_cloth], axis=1)
+    net.setInput(inp)
+    net.setPreferableBackend(backend)
+    net.setPreferableTarget(target)
+    out = net.forward()
+
+    p_rendered, m_composite = np.split(out, [3], axis=1)
+    p_rendered = np.tanh(p_rendered)
+    m_composite = sigmoid(m_composite)
```

Comment:
Can you compute sigmoid with just numpy?

Reply:
mb overflow

Reply:
do you mean in case of `exp(x)` in `1 / (1 + exp(-x))` ?